# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/util/ActiveElementManager.h
# Commit: 88bd13bb33d1
# Full Hash: 88bd13bb33d1d623619e8956cc71cde0e3797a4d
# Author: Dan Robertson <drobertson@mozilla.com>
# Date: 2023-08-24 14:56:02
# Regressor Bug: 1816473
# File Overlap Count: 1
# Description:
#   Bug 1816473 - Do not delay dispatching synthesized mouse events. r=botond,edgar
#   
#   Do not delay dispatching synthesized mouse events for a single tap
#   gesture.
#   
# ==============================================================================

diff -r 1d5eb1343cc8 -r 88bd13bb33d1 gfx/layers/apz/util/ActiveElementManager.h
--- a/gfx/layers/apz/util/ActiveElementManager.h	Wed Aug 23 19:53:47 2023 -0400
+++ b/gfx/layers/apz/util/ActiveElementManager.h	Wed Aug 23 23:57:25 2023 +0000
@@ -21,6 +21,8 @@
 
 namespace layers {
 
+class DelayedClearElementActivation;
+
 /**
  * Manages setting and clearing the ':active' CSS pseudostate in the presence
  * of touch input.
@@ -61,6 +63,10 @@
    * delayed until after touch listeners have responded to the APZ.
    */
   void HandleTouchEnd();
+  /**
+   * Possibly clear active element sate in response to a single tap.
+   */
+  void ProcessSingleTap();
 
  private:
   /**
@@ -82,6 +88,10 @@
    */
   RefPtr<CancelableRunnable> mSetActiveTask;
 
+  // Store the pending single tap event element activation clearing
+  // task.
+  RefPtr<DelayedClearElementActivation> mDelayedClearElementActivation;
+
   // Helpers
   void TriggerElementActivation();
   void SetActive(dom::Element* aTarget);
